### ISSUE TITLE
Long shader compilation time for Android

### ISSUE THREAD

**Issue by spirosperos** [KhronosGroup/UnityGLTF#802]
Hello! I'm working on a project involving a Quest 3s (Android) device and Unity. I am currently trying to load a .glb file and visualize it in VR. I'm facing a problem regarding missing shaders, which should be possible to solve by adding UnityGLTF/PBRGraph to the list of always included shaders. However, when I do this I get the following error:

`Error building Player: UnityGLTF/PBRGraph has too many Shader variants (1179648)`

After taking a look at the issues, I've found this [comment](https://github.com/KhronosGroup/UnityGLTF/issues/736#issuecomment-2416004030) saying that I should try preloading the predefined shader variant collection. This leads to a huge compilation time, due to a big number of variants.

![image](https://github.com/user-attachments/assets/9d07706f-f701-40f1-8e43-12fe5cb4378b)

Is there anything I could do to speed this process up? I'm building using Unity 2022.3.8f1

**Comment by hybridherbst** [comment]
Hey, couple questions:

- Which UnityGLTF version are you using?
- Are you using URP or BiRP?
- Are you using any specific renderer features in your URP asset?
- Did you remove the shader from the "always included shaders" list after adding the premade variant sets?

**Comment by atkulp** [comment]
This is what worked for me. Don't add it to Always Included, add the variant collection to Preloaded Shaders. Just match the variant collection to the pipeline setting.

![Image](https://github.com/user-attachments/assets/af78b962-c815-4bfc-aa46-e8ed552db185)

**Comment by hybridherbst** [comment]
Yes, using the provided Shader Variant Collections is the suggested resolution here. Adding to "Always included" will cause Unity to compile a giant amount of unnecessary variants, which they told me is "by design" unfortunately.

Please use the Variant Collection! And if that doesn't work, please open a new issue.
